### PR TITLE
Adds handling for Aeon requests

### LINKED ISSUE #17: Deliver Reading Room request to Aeon
## Describe the solution you'd like
Serialize and deliver data from submitted list to Aeon. Data should include all information necessary to create a request in Aeon.

This action should be triggered by a POST request to a specific endpoint which contains request data, and should in turn deliver a POST request with this data as `application/x-www-form-urlencoded` data to an Aeon endpoint. A response to the initial post request should be returned which indicates whether the delivery succeeded or failed.

### LINKED ISSUE #18: Deliver Duplication requests to Aeon
## Describe the solution you'd like
Serialize and deliver data from submitted list to Aeon. Data should include all information necessary to create a request in Aeon.

This action should be triggered by a POST request to a specific endpoint which contains request data, and should in turn deliver a POST request with this data as `application/x-www-form-urlencoded` data to an Aeon endpoint. A response to the initial post request should be returned which indicates whether the delivery succeeded or failed.

### PR DESCRIPTION
**Pull request by helrond**
This was originally intended to be targeted just at Reading Room requests, but once I got into it, I realized that Duplication requests would basically use the same logic, so I've included code for that in this PR as well. 
fixes #17 
fixes #18 

Some notes:
- There are a couple of config updates, so you'll have to recreate `config.py` from `config.py.example` or add values to your existing file.
- We have been promised an Aeon API endpoint for creating requests, but that doesn't exist yet, so it's not possible to fully test these methods. Changes to the structure of submitted data (among other things) may be necessary.
- The AeonAPIClient code is based on ArchivesSnake. Depending on how idiosyncratic the Aeon API turns out to be, we could consider turning this into an installable package for other people to use.

### CODE REVIEW

**Comment by p-galligan** [comment]
@helrond There are obviously some TODOs in the parsing section of creating the AEON API requests and that's related to issue #53. I'm assuming we're good merging this PR as-is and then expanding the AS data as necessary to include the work in #53?

**Comment by helrond** [comment]
@p-galligan that was my thought, especially since we can't really test against an Aeon endpoint right now.

**Review by p-galligan — APPROVED** [review]

File: process_request/views.py [hunk 1]
```diff
@@ -47,6 +47,32 @@ def post(self, request):
             return Response({"detail": str(e)}, status=500)
 
 
+class DeliverReadingRoomRequestView(APIView):
+    "Delivers a request for records to be delivered to the reading room."
```
Comment (bonniegee) [inline process_request/views.py]: Should this and line 64 be comments?

Comment (p-galligan) [reply process_request/views.py]: Oh, good catch. This should actually probably be docstrings, right @helrond?

Comment (helrond) [reply process_request/views.py]: Good catch - yup, those are docstrings. Updated.

File: process_request/clients.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from request_broker import settings
+from requests import Session
+
+
+def http_meth_factory(meth):
+    """Utility method for producing HTTP proxy methods.
+
+    Urls are prefixed with the value of baseurl from the client's config.
+    Arguments are passed unaltered to the matching requests.Session method."""
```
Comment (HaSistrunk) [inline process_request/clients.py:9]:
```suggestion
    Arguments are passed unaltered to the matching requests. Session method."""
```

File: process_request/views.py [hunk 3]
```diff
@@ -47,6 +47,32 @@ def post(self, request):
...
+            request_data = request.data
+            delivered = AeonRequester().send_request(
+                "readingroom", **request_data)
+            return Response({"detail": delivered}, status=200)
+        except Exception as e:
+            return Response({"detail": str(e)}, status=500)
+
+
+class DeliverDuplicationRequestView(APIView):
+    """Delivers a request for records to be delivered to the reading room."""
```
Comment (HaSistrunk) [inline process_request/views.py]:
```suggestion
    """Delivers a request for records to be duplicated."""
```

**Review by HaSistrunk — APPROVED** [review]

File: process_request/clients.py (hunk 2 above)
Comment (helrond) [reply process_request/clients.py:9]: This is actually right - `requests.Session` refers to a `requests` class called `Session`.

**Review by bonniegee — APPROVED** [review]

**Pull request merged**